src/node.cc

```diff
@@ -1269,41 +1264,22 @@ Local<Value> MakeCallback(Environment* env,
     }
   }
 
-  if (ran_init_callback && !pre_fn.IsEmpty()) {
-    TryCatch try_catch(env->isolate());
-    MaybeLocal<Value> ar = pre_fn->Call(env->context(), object, 0, nullptr);
-    if (ar.IsEmpty()) {
-      ClearFatalExceptionHandlers(env);
-      FatalException(env->isolate(), try_catch);
-      return Local<Value>();
-    }
-  }
+  // TODO(trevnorris): Correct this once node::MakeCallback() support id and
+  // triggerId. Consider completely removing it until then so the async id can
+  // propagate through to the fatalException after hook calls.
```

Comment:
I don't think a fix can be made, we will just have to default to the null/root-parent. If a fix did exist, async-wrap shouldn't be necessary.

However, I do believe we can migrate quite well using NAN, since that wraps the callback upon entering C++, to make it persistent. Thus we can create an asyncId when it wraps, and call the new `MakeCallback` when `wrap->Call` is executed.

If I remember correctly it also has a cleanup call we can use for calling `destroy`.